src/MongoDB.Driver.Core/Core/Operations/RetryableDeleteCommandOperation.cs

```diff
@@ -128,6 +137,11 @@ protected override void SerializeValue(BsonSerializationContext context, BsonSer
                 writer.WriteStartDocument();
                 writer.WriteName("q");
                 BsonDocumentSerializer.Instance.Serialize(context, value.Filter);
+                if (value.Hint != null)
+                {
+                    writer.WriteName("hint");
+                    BsonValueSerializer.Instance.Serialize(context, value.Hint);
+                }
```

Comment:
Move these new line down a bit so that `hint` is output *after* `collation` (like it is in the server documentation).

Reply:
Done.
For this change I also had to add "deletes" operation to order-agnostic commands here: [link](https://github.com/mongodb/mongo-csharp-driver/blob/1bf7f5326e49c2f220aab6a38217edaf2daa5d26/tests/MongoDB.Driver.Core.TestHelpers/JsonDrivenTests/CommandStartedEventAsserter.cs#L153), because in spec "hint" is before "limit" ([link](https://github.com/mongodb/specifications/blob/4400c6c347e46bc51e063add1a8cbf05399285b8/source/crud/tests/v2/bulkWrite-delete-hint.json#L181))

Reply:
Interesting. I was suggesting the order used in the server documentation:

https://docs.mongodb.com/master/reference/command/delete/